## Import Data

In [1]:
import pandas as pd

In [2]:
pd.set_option('display.max_rows', None)
dataframe = pd.read_csv('data-science.csv')
dataframe

Unnamed: 0                                       Company Name  Rating  \
0             0                                     Jun Group\n4.2     4.2   
1             1                      Liberty Mutual Insurance\n3.9     3.9   
2             2                                       the NBA\n4.2     4.2   
3             3                                  EXL Services\n3.7     3.7   
4             4                                         KAYAK\n4.4     4.4   
5             5                                          Bose\n3.8     3.8   
6             6                                    Kaizen Dynamics     NaN   
7             7                                    LexisNexis\n3.9     3.9   
8             8                                          Visa\n4.0     4.0   
9             9                              Longevity InTime\n3.9     3.9   
10           10                                           DSMH LLC     NaN   
11           11                         Fathom Management LLC\n2.0     2.0   
12           12                                          Wish\n3.3     3.3   
13           13                       Uptake Technologies Inc\n3.8     3.8   
14           14                                         Zelis\n3.6     3.6   
15           15                                         Apple\n4.2     4.2   
16           16                                       PepsiCo\n3.9     3.9   
17           17                                     Silverxis\n4.0     4.0   
18           18                                         Net2Aspire     NaN   
19           19                                    MasterCard\n4.3     4.3   
20           20                                           Dworo.io     NaN   
21           21                                          RELX\n4.0     4.0   
22           22                                          Dataworks     NaN   
23           23                           Booz Allen Hamilton\n4.2     4.2   
24           24        University of California San Francisco\n4.1     4.1   
25           25                                   Point1 Solutions     NaN   
26           26     International Consulting Associates, Inc.\n5.0     5.0   
27           27                            Brilliant Infotech\n3.5     3.5   
28           28                            PlayStation Global\n3.9     3.9   
29           29                                          Yelp\n3.5     3.5   
30           30                                          Bose\n3.8     3.8   
31           31                      Liberty Mutual Insurance\n3.9     3.9   
32           32                                     Jun Group\n4.2     4.2   
33           33                                  EXL Services\n3.7     3.7   
34           34                                       the NBA\n4.2     4.2   
35           35                                     Heading Health     NaN   
36           36                                          Visa\n4.0     4.0   
37           37                                    LexisNexis\n3.9     3.9   
38           38                                         KAYAK\n4.4     4.4   
39           39                                         Net2Aspire     NaN   
40           40                                           Dworo.io     NaN   
41           41                                    MasterCard\n4.3     4.3   
42           42                                         Apple\n4.2     4.2   
43           43                                         Zelis\n3.6     3.6   
44           44                                          Wish\n3.3     3.3   
45           45                                          RELX\n4.0     4.0   
46           46                UT Southwestern Medical Center\n3.9     3.9   
47           47                                       Skit.ai\n4.5     4.5   
48           48                                        WeLink\n4.4     4.4   
49           49                         Fathom Management LLC\n2.0     2.0   
50           50            

In [3]:
dataframe.shape

(900, 13)

## Clean the duplicate value

I noticed that glasssdoor repeated the same job title in the same company on the search job results and it makes the data repeated when I scrape it. So I cleaned it by dropping the duplicated values on the subset Company Name, Job Title, and Job Description

In [4]:
df = dataframe.drop_duplicates(subset =['Company Name','Job Title','Job Description'], keep = 'first', inplace = False)

In [5]:
df.shape

(242, 13)

In [6]:
df.duplicated().sum()

0

Now, the dataset has 0 duplicate values and have 242 row and 13 columns.

## Salary Parsing

Remove the values that has NaN on the Salary Estimate column

In [7]:
df = df[df['Salary Estimate'].notna()]
df

Unnamed: 0                                       Company Name  Rating  \
0             0                                     Jun Group\n4.2     4.2   
1             1                      Liberty Mutual Insurance\n3.9     3.9   
2             2                                       the NBA\n4.2     4.2   
4             4                                         KAYAK\n4.4     4.4   
6             6                                    Kaizen Dynamics     NaN   
7             7                                    LexisNexis\n3.9     3.9   
8             8                                          Visa\n4.0     4.0   
9             9                              Longevity InTime\n3.9     3.9   
10           10                                           DSMH LLC     NaN   
11           11                         Fathom Management LLC\n2.0     2.0   
12           12                                          Wish\n3.3     3.3   
16           16                                       PepsiCo\n3.9     3.9   
17           17                                     Silverxis\n4.0     4.0   
18           18                                         Net2Aspire     NaN   
19           19                                    MasterCard\n4.3     4.3   
20           20                                           Dworo.io     NaN   
21           21                                          RELX\n4.0     4.0   
22           22                                          Dataworks     NaN   
23           23                           Booz Allen Hamilton\n4.2     4.2   
24           24        University of California San Francisco\n4.1     4.1   
27           27                            Brilliant Infotech\n3.5     3.5   
28           28                            PlayStation Global\n3.9     3.9   
29           29                                          Yelp\n3.5     3.5   
31           31                      Liberty Mutual Insurance\n3.9     3.9   
32           32                                     Jun Group\n4.2     4.2   
35           35                                     Heading Health     NaN   
36           36                                          Visa\n4.0     4.0   
38           38                                         KAYAK\n4.4     4.4   
41           41                                    MasterCard\n4.3     4.3   
44           44                                          Wish\n3.3     3.3   
46           46                UT Southwestern Medical Center\n3.9     3.9   
47           47                                       Skit.ai\n4.5     4.5   
51           51                           Booz Allen Hamilton\n4.2     4.2   
53           53                                      Deloitte\n4.0     4.0   
55           55                           Pearl Certification\n4.7     4.7   
56           56                                          Nike\n4.1     4.1   
57           57                                       PepsiCo\n3.9     3.9   
66           66                                          Visa\n4.0     4.0   
70           70                           Booz Allen Hamilton\n4.2     4.2   
73           73                                         Roche\n4.2     4.2   
79           79                                       Spotify\n4.2     4.2   
80           80        University of California San Francisco\n4.1     4.1   
88           88                                       KeyBank\n3.8     3.8   
91           91                      Liberty Mutual Insurance\n3.9     3.9   
101         101                UT Southwestern Medical Center\n3.9     3.9   
103         103                                    MasterCard\n4.3     4.3   
111         111                           Booz Allen Hamilton\n4.2     4.2   
118         118                                    Zelus Analytics     NaN   
139         139                                      Deloitte\n4.0     4.0   
144         144                                     Starbucks\n3.8     3.8   
146         146            

append the hourly and employer provided salary to calculate the annual rate from per hour

In [8]:
df['hourly'] = df['Salary Estimate'].apply(lambda x: 1 if 'per hour' in x.lower() else 0)
df['employer_provided'] = df['Salary Estimate'].apply(lambda x: 1 if 'employer provided salary' in x.lower() else 0)
df

Unnamed: 0                                       Company Name  Rating  \
0             0                                     Jun Group\n4.2     4.2   
1             1                      Liberty Mutual Insurance\n3.9     3.9   
2             2                                       the NBA\n4.2     4.2   
4             4                                         KAYAK\n4.4     4.4   
6             6                                    Kaizen Dynamics     NaN   
7             7                                    LexisNexis\n3.9     3.9   
8             8                                          Visa\n4.0     4.0   
9             9                              Longevity InTime\n3.9     3.9   
10           10                                           DSMH LLC     NaN   
11           11                         Fathom Management LLC\n2.0     2.0   
12           12                                          Wish\n3.3     3.3   
16           16                                       PepsiCo\n3.9     3.9   
17           17                                     Silverxis\n4.0     4.0   
18           18                                         Net2Aspire     NaN   
19           19                                    MasterCard\n4.3     4.3   
20           20                                           Dworo.io     NaN   
21           21                                          RELX\n4.0     4.0   
22           22                                          Dataworks     NaN   
23           23                           Booz Allen Hamilton\n4.2     4.2   
24           24        University of California San Francisco\n4.1     4.1   
27           27                            Brilliant Infotech\n3.5     3.5   
28           28                            PlayStation Global\n3.9     3.9   
29           29                                          Yelp\n3.5     3.5   
31           31                      Liberty Mutual Insurance\n3.9     3.9   
32           32                                     Jun Group\n4.2     4.2   
35           35                                     Heading Health     NaN   
36           36                                          Visa\n4.0     4.0   
38           38                                         KAYAK\n4.4     4.4   
41           41                                    MasterCard\n4.3     4.3   
44           44                                          Wish\n3.3     3.3   
46           46                UT Southwestern Medical Center\n3.9     3.9   
47           47                                       Skit.ai\n4.5     4.5   
51           51                           Booz Allen Hamilton\n4.2     4.2   
53           53                                      Deloitte\n4.0     4.0   
55           55                           Pearl Certification\n4.7     4.7   
56           56                                          Nike\n4.1     4.1   
57           57                                       PepsiCo\n3.9     3.9   
66           66                                          Visa\n4.0     4.0   
70           70                           Booz Allen Hamilton\n4.2     4.2   
73           73                                         Roche\n4.2     4.2   
79           79                                       Spotify\n4.2     4.2   
80           80        University of California San Francisco\n4.1     4.1   
88           88                                       KeyBank\n3.8     3.8   
91           91                      Liberty Mutual Insurance\n3.9     3.9   
101         101                UT Southwestern Medical Center\n3.9     3.9   
103         103                                    MasterCard\n4.3     4.3   
111         111                           Booz Allen Hamilton\n4.2     4.2   
118         118                                    Zelus Analytics     NaN   
139         139                                      Deloitte\n4.0     4.0   
144         144                                     Starbucks\n3.8     3.8   
146         146            

Remove the text from field 'Salary Estimate' that says 'per hour' and 'emoloyer provided salary', parenthesis, remove the K and dollar sign

In [9]:
salary = df['Salary Estimate'].apply(lambda x: x.split('(')[0])
minus_Kd = salary.apply(lambda x: x.replace('K','').replace('$',''))
min_hrPS = minus_Kd.apply(lambda x: x.lower().replace('per hour','').replace('employer provided salary:',''))
min_hrPS

0       80 - 126 
1        99 - 115
2       97 - 141 
4       91 - 124 
6        40 - 50 
7       67 - 104 
8      106 - 146 
9       100 - 150
10       60 - 65 
11      110 - 130
12     127 - 180 
16      70 - 104 
17            25 
18        65 - 80
19      81 - 114 
20        76 - 88
21       64 - 94 
22      115 - 130
23      78 - 119 
24      97 - 137 
27       50 - 60 
28      94 - 138 
29       88 - 220
31       99 - 115
32      80 - 126 
35        50 - 65
36     106 - 146 
38      91 - 124 
41      85 - 119 
44     127 - 180 
46       67 - 120
47       38 - 58 
51      77 - 115 
53       46 - 73 
55       90 - 130
56     100 - 142 
57      70 - 104 
66     106 - 146 
70      77 - 115 
73       90 - 135
79      93 - 134 
80      67 - 100 
88       70 - 110
91       99 - 115
101      67 - 120
103     81 - 114 
111     77 - 115 
118      75 - 150
139      46 - 73 
144      61 - 96 
146     86 - 119 
147     66 - 113 
158      61 - 96 
174     66 - 113 
193     70 - 104 
205     66

add min_salary, max_salary, and avg_salary columns

In [10]:
df['min_salary'] = min_hrPS.apply(lambda x: int(x.split('-')[0]) if '-' in x else x)
df['max_salary'] = min_hrPS.apply(lambda x: int(x.split('-')[1]) if '-' in x else x)
df['avg_salary'] = (df.min_salary.astype(int) + df.max_salary.astype(int))/2
df

Unnamed: 0                                       Company Name  Rating  \
0             0                                     Jun Group\n4.2     4.2   
1             1                      Liberty Mutual Insurance\n3.9     3.9   
2             2                                       the NBA\n4.2     4.2   
4             4                                         KAYAK\n4.4     4.4   
6             6                                    Kaizen Dynamics     NaN   
7             7                                    LexisNexis\n3.9     3.9   
8             8                                          Visa\n4.0     4.0   
9             9                              Longevity InTime\n3.9     3.9   
10           10                                           DSMH LLC     NaN   
11           11                         Fathom Management LLC\n2.0     2.0   
12           12                                          Wish\n3.3     3.3   
16           16                                       PepsiCo\n3.9     3.9   
17           17                                     Silverxis\n4.0     4.0   
18           18                                         Net2Aspire     NaN   
19           19                                    MasterCard\n4.3     4.3   
20           20                                           Dworo.io     NaN   
21           21                                          RELX\n4.0     4.0   
22           22                                          Dataworks     NaN   
23           23                           Booz Allen Hamilton\n4.2     4.2   
24           24        University of California San Francisco\n4.1     4.1   
27           27                            Brilliant Infotech\n3.5     3.5   
28           28                            PlayStation Global\n3.9     3.9   
29           29                                          Yelp\n3.5     3.5   
31           31                      Liberty Mutual Insurance\n3.9     3.9   
32           32                                     Jun Group\n4.2     4.2   
35           35                                     Heading Health     NaN   
36           36                                          Visa\n4.0     4.0   
38           38                                         KAYAK\n4.4     4.4   
41           41                                    MasterCard\n4.3     4.3   
44           44                                          Wish\n3.3     3.3   
46           46                UT Southwestern Medical Center\n3.9     3.9   
47           47                                       Skit.ai\n4.5     4.5   
51           51                           Booz Allen Hamilton\n4.2     4.2   
53           53                                      Deloitte\n4.0     4.0   
55           55                           Pearl Certification\n4.7     4.7   
56           56                                          Nike\n4.1     4.1   
57           57                                       PepsiCo\n3.9     3.9   
66           66                                          Visa\n4.0     4.0   
70           70                           Booz Allen Hamilton\n4.2     4.2   
73           73                                         Roche\n4.2     4.2   
79           79                                       Spotify\n4.2     4.2   
80           80        University of California San Francisco\n4.1     4.1   
88           88                                       KeyBank\n3.8     3.8   
91           91                      Liberty Mutual Insurance\n3.9     3.9   
101         101                UT Southwestern Medical Center\n3.9     3.9   
103         103                                    MasterCard\n4.3     4.3   
111         111                           Booz Allen Hamilton\n4.2     4.2   
118         118                                    Zelus Analytics     NaN   
139         139                                      Deloitte\n4.0     4.0   
144         144                                     Starbucks\n3.8     3.8   
146         146            

## Company Name Text Only

In [11]:
df['company_txt'] = df.apply(lambda x: x['Company Name'] if x['Rating']<0 else x['Company Name'][:-4], axis = 1)
df

Unnamed: 0                                       Company Name  Rating  \
0             0                                     Jun Group\n4.2     4.2   
1             1                      Liberty Mutual Insurance\n3.9     3.9   
2             2                                       the NBA\n4.2     4.2   
4             4                                         KAYAK\n4.4     4.4   
6             6                                    Kaizen Dynamics     NaN   
7             7                                    LexisNexis\n3.9     3.9   
8             8                                          Visa\n4.0     4.0   
9             9                              Longevity InTime\n3.9     3.9   
10           10                                           DSMH LLC     NaN   
11           11                         Fathom Management LLC\n2.0     2.0   
12           12                                          Wish\n3.3     3.3   
16           16                                       PepsiCo\n3.9     3.9   
17           17                                     Silverxis\n4.0     4.0   
18           18                                         Net2Aspire     NaN   
19           19                                    MasterCard\n4.3     4.3   
20           20                                           Dworo.io     NaN   
21           21                                          RELX\n4.0     4.0   
22           22                                          Dataworks     NaN   
23           23                           Booz Allen Hamilton\n4.2     4.2   
24           24        University of California San Francisco\n4.1     4.1   
27           27                            Brilliant Infotech\n3.5     3.5   
28           28                            PlayStation Global\n3.9     3.9   
29           29                                          Yelp\n3.5     3.5   
31           31                      Liberty Mutual Insurance\n3.9     3.9   
32           32                                     Jun Group\n4.2     4.2   
35           35                                     Heading Health     NaN   
36           36                                          Visa\n4.0     4.0   
38           38                                         KAYAK\n4.4     4.4   
41           41                                    MasterCard\n4.3     4.3   
44           44                                          Wish\n3.3     3.3   
46           46                UT Southwestern Medical Center\n3.9     3.9   
47           47                                       Skit.ai\n4.5     4.5   
51           51                           Booz Allen Hamilton\n4.2     4.2   
53           53                                      Deloitte\n4.0     4.0   
55           55                           Pearl Certification\n4.7     4.7   
56           56                                          Nike\n4.1     4.1   
57           57                                       PepsiCo\n3.9     3.9   
66           66                                          Visa\n4.0     4.0   
70           70                           Booz Allen Hamilton\n4.2     4.2   
73           73                                         Roche\n4.2     4.2   
79           79                                       Spotify\n4.2     4.2   
80           80        University of California San Francisco\n4.1     4.1   
88           88                                       KeyBank\n3.8     3.8   
91           91                      Liberty Mutual Insurance\n3.9     3.9   
101         101                UT Southwestern Medical Center\n3.9     3.9   
103         103                                    MasterCard\n4.3     4.3   
111         111                           Booz Allen Hamilton\n4.2     4.2   
118         118                                    Zelus Analytics     NaN   
139         139                                      Deloitte\n4.0     4.0   
144         144                                     Starbucks\n3.8     3.8   
146         146            

## State Field

In [12]:
df['job_state'] = df['Location'].apply(lambda x: x.split(',')[1] if ',' in x else x)
df.job_state.value_counts()

Remote       34
 CA          33
 NY          24
 TX          16
 WA          12
 PA          11
 VA          10
 CO          10
 MO           7
 IL           6
 MD           5
 UT           5
 DC           4
 IN           4
 FL           4
 NC           3
 MA           3
 GA           2
 IA           2
 NJ           2
 TN           2
 NE           2
Maryland      1
Manhattan     1
 RI           1
 KS           1
 WI           1
 MN           1
Tennessee     1
 CT           1
Name: job_state, dtype: int64

## Age of Company

In [13]:
df['age'] = df['Founded'].apply(lambda x: x if x <1 else 2022-x)
df

Unnamed: 0                                       Company Name  Rating  \
0             0                                     Jun Group\n4.2     4.2   
1             1                      Liberty Mutual Insurance\n3.9     3.9   
2             2                                       the NBA\n4.2     4.2   
4             4                                         KAYAK\n4.4     4.4   
6             6                                    Kaizen Dynamics     NaN   
7             7                                    LexisNexis\n3.9     3.9   
8             8                                          Visa\n4.0     4.0   
9             9                              Longevity InTime\n3.9     3.9   
10           10                                           DSMH LLC     NaN   
11           11                         Fathom Management LLC\n2.0     2.0   
12           12                                          Wish\n3.3     3.3   
16           16                                       PepsiCo\n3.9     3.9   
17           17                                     Silverxis\n4.0     4.0   
18           18                                         Net2Aspire     NaN   
19           19                                    MasterCard\n4.3     4.3   
20           20                                           Dworo.io     NaN   
21           21                                          RELX\n4.0     4.0   
22           22                                          Dataworks     NaN   
23           23                           Booz Allen Hamilton\n4.2     4.2   
24           24        University of California San Francisco\n4.1     4.1   
27           27                            Brilliant Infotech\n3.5     3.5   
28           28                            PlayStation Global\n3.9     3.9   
29           29                                          Yelp\n3.5     3.5   
31           31                      Liberty Mutual Insurance\n3.9     3.9   
32           32                                     Jun Group\n4.2     4.2   
35           35                                     Heading Health     NaN   
36           36                                          Visa\n4.0     4.0   
38           38                                         KAYAK\n4.4     4.4   
41           41                                    MasterCard\n4.3     4.3   
44           44                                          Wish\n3.3     3.3   
46           46                UT Southwestern Medical Center\n3.9     3.9   
47           47                                       Skit.ai\n4.5     4.5   
51           51                           Booz Allen Hamilton\n4.2     4.2   
53           53                                      Deloitte\n4.0     4.0   
55           55                           Pearl Certification\n4.7     4.7   
56           56                                          Nike\n4.1     4.1   
57           57                                       PepsiCo\n3.9     3.9   
66           66                                          Visa\n4.0     4.0   
70           70                           Booz Allen Hamilton\n4.2     4.2   
73           73                                         Roche\n4.2     4.2   
79           79                                       Spotify\n4.2     4.2   
80           80        University of California San Francisco\n4.1     4.1   
88           88                                       KeyBank\n3.8     3.8   
91           91                      Liberty Mutual Insurance\n3.9     3.9   
101         101                UT Southwestern Medical Center\n3.9     3.9   
103         103                                    MasterCard\n4.3     4.3   
111         111                           Booz Allen Hamilton\n4.2     4.2   
118         118                                    Zelus Analytics     NaN   
139         139                                      Deloitte\n4.0     4.0   
144         144                                     Starbucks\n3.8     3.8   
146         146            

## Parsing of Job Description

Popular tools that needed by the Data Scientist

1. Python

In [14]:
df['python'] = df['Job Description'].apply(lambda x: 1 if 'python' in x.lower() else 0)
df.python.value_counts()

1    153
0     56
Name: python, dtype: int64

2. R Studio

In [15]:
df['r_studio'] = df['Job Description'].apply(lambda x: 1 if 'r studio' in x.lower() or ' r ' in x.lower() else 0)
df.r_studio.value_counts()

0    167
1     42
Name: r_studio, dtype: int64

3. Spark

In [16]:
df['spark'] = df['Job Description'].apply(lambda x: 1 if 'spark' in x.lower() else 0)
df.spark.value_counts()

0    182
1     27
Name: spark, dtype: int64

4. AWS

In [17]:
df['aws'] = df['Job Description'].apply(lambda x: 1 if 'aws' in x.lower() else 0)
df.aws.value_counts()

0    151
1     58
Name: aws, dtype: int64

5. Excel

In [18]:
df['excel'] = df['Job Description'].apply(lambda x: 1 if 'excel' in x.lower() else 0)
df.excel.value_counts()

0    115
1     94
Name: excel, dtype: int64

In [19]:
df_out = df.drop(['Unnamed: 0'], axis = 1)
df_out

Company Name  Rating  \
0                                       Jun Group\n4.2     4.2   
1                        Liberty Mutual Insurance\n3.9     3.9   
2                                         the NBA\n4.2     4.2   
4                                           KAYAK\n4.4     4.4   
6                                      Kaizen Dynamics     NaN   
7                                      LexisNexis\n3.9     3.9   
8                                            Visa\n4.0     4.0   
9                                Longevity InTime\n3.9     3.9   
10                                            DSMH LLC     NaN   
11                          Fathom Management LLC\n2.0     2.0   
12                                           Wish\n3.3     3.3   
16                                        PepsiCo\n3.9     3.9   
17                                      Silverxis\n4.0     4.0   
18                                          Net2Aspire     NaN   
19                                     MasterCard\n4.3     4.3   
20                                            Dworo.io     NaN   
21                                           RELX\n4.0     4.0   
22                                           Dataworks     NaN   
23                            Booz Allen Hamilton\n4.2     4.2   
24         University of California San Francisco\n4.1     4.1   
27                             Brilliant Infotech\n3.5     3.5   
28                             PlayStation Global\n3.9     3.9   
29                                           Yelp\n3.5     3.5   
31                       Liberty Mutual Insurance\n3.9     3.9   
32                                      Jun Group\n4.2     4.2   
35                                      Heading Health     NaN   
36                                           Visa\n4.0     4.0   
38                                          KAYAK\n4.4     4.4   
41                                     MasterCard\n4.3     4.3   
44                                           Wish\n3.3     3.3   
46                 UT Southwestern Medical Center\n3.9     3.9   
47                                        Skit.ai\n4.5     4.5   
51                            Booz Allen Hamilton\n4.2     4.2   
53                                       Deloitte\n4.0     4.0   
55                            Pearl Certification\n4.7     4.7   
56                                           Nike\n4.1     4.1   
57                                        PepsiCo\n3.9     3.9   
66                                           Visa\n4.0     4.0   
70                            Booz Allen Hamilton\n4.2     4.2   
73                                          Roche\n4.2     4.2   
79                                        Spotify\n4.2     4.2   
80         University of California San Francisco\n4.1     4.1   
88                                        KeyBank\n3.8     3.8   
91                       Liberty Mutual Insurance\n3.9     3.9   
101                UT Southwestern Medical Center\n3.9     3.9   
103                                    MasterCard\n4.3     4.3   
111                           Booz Allen Hamilton\n4.2     4.2   
118                                    Zelus Analytics     NaN   
139                                      Deloitte\n4.0     4.0   
144                                     Starbucks\n3.8     3.8   
146                                         Bayer\n4.1     4.1   
147  Andela - Third Party Job Board Only Postings\n3.9     3.9   
158                                     Starbucks\n3.8     3.8   
174  Andela - Third Party Job Board Only Postings\n3.9     3.9   
193                                       PepsiCo\n3.9     3.9   
205  Andela - Third Party Job Board Only Postings\n3.9     3.9   
223                                    MasterCard\n4.3     4.3   
233                                         Bayer\n4.1     4.1   
247                                     Starbucks\n3.8     3.8   
262                                      Deloitte\n4.0     4.0   
263             

In [20]:
df_out.to_csv('Salary Data Cleaned.csv', index = False)

In [21]:
pd.read_csv('Salary Data Cleaned.csv')

Company Name  Rating  \
0                                       Jun Group\n4.2     4.2   
1                        Liberty Mutual Insurance\n3.9     3.9   
2                                         the NBA\n4.2     4.2   
3                                           KAYAK\n4.4     4.4   
4                                      Kaizen Dynamics     NaN   
5                                      LexisNexis\n3.9     3.9   
6                                            Visa\n4.0     4.0   
7                                Longevity InTime\n3.9     3.9   
8                                             DSMH LLC     NaN   
9                           Fathom Management LLC\n2.0     2.0   
10                                           Wish\n3.3     3.3   
11                                        PepsiCo\n3.9     3.9   
12                                      Silverxis\n4.0     4.0   
13                                          Net2Aspire     NaN   
14                                     MasterCard\n4.3     4.3   
15                                            Dworo.io     NaN   
16                                           RELX\n4.0     4.0   
17                                           Dataworks     NaN   
18                            Booz Allen Hamilton\n4.2     4.2   
19         University of California San Francisco\n4.1     4.1   
20                             Brilliant Infotech\n3.5     3.5   
21                             PlayStation Global\n3.9     3.9   
22                                           Yelp\n3.5     3.5   
23                       Liberty Mutual Insurance\n3.9     3.9   
24                                      Jun Group\n4.2     4.2   
25                                      Heading Health     NaN   
26                                           Visa\n4.0     4.0   
27                                          KAYAK\n4.4     4.4   
28                                     MasterCard\n4.3     4.3   
29                                           Wish\n3.3     3.3   
30                 UT Southwestern Medical Center\n3.9     3.9   
31                                        Skit.ai\n4.5     4.5   
32                            Booz Allen Hamilton\n4.2     4.2   
33                                       Deloitte\n4.0     4.0   
34                            Pearl Certification\n4.7     4.7   
35                                           Nike\n4.1     4.1   
36                                        PepsiCo\n3.9     3.9   
37                                           Visa\n4.0     4.0   
38                            Booz Allen Hamilton\n4.2     4.2   
39                                          Roche\n4.2     4.2   
40                                        Spotify\n4.2     4.2   
41         University of California San Francisco\n4.1     4.1   
42                                        KeyBank\n3.8     3.8   
43                       Liberty Mutual Insurance\n3.9     3.9   
44                 UT Southwestern Medical Center\n3.9     3.9   
45                                     MasterCard\n4.3     4.3   
46                            Booz Allen Hamilton\n4.2     4.2   
47                                     Zelus Analytics     NaN   
48                                       Deloitte\n4.0     4.0   
49                                      Starbucks\n3.8     3.8   
50                                          Bayer\n4.1     4.1   
51   Andela - Third Party Job Board Only Postings\n3.9     3.9   
52                                      Starbucks\n3.8     3.8   
53   Andela - Third Party Job Board Only Postings\n3.9     3.9   
54                                        PepsiCo\n3.9     3.9   
55   Andela - Third Party Job Board Only Postings\n3.9     3.9   
56                                     MasterCard\n4.3     4.3   
57                                          Bayer\n4.1     4.1   
58                                      Starbucks\n3.8     3.8   
59                                       Deloitte\n4.0     4.0   
60              